net_io.c

```diff
@@ -776,13 +782,270 @@ static void send_sbs_heartbeat(struct net_service *service)
+    } else if (mm->msgtype == 18) {
+        cacf = mm->CF;
+    }
+
+    p += sprintf(p,
+            "{\"Icao_addr\":%d,"
+            "\"DF\":%d,\"CA\":%d,"
+            "\"TypeCode\":%d,"
+            "\"SubtypeCode\":%d,"
+            "\"SBS_MsgType\":%d,"
+            "\"SignalLevel\":%f,",
+            mm->addr,
+            mm->msgtype, cacf,
+            mm->metype,
+            mm->mesub,
+            msgType,
+            mm->signalLevel); // what precision and range is needed for RSSI?
+
+    // Find current system time
+    clock_gettime(CLOCK_REALTIME, &now);
+    gmtime_r(&now.tv_sec, &stTime_now);  // we expect UTC
+
+    // Find message reception time
+    time_t received = (time_t) (mm->sysTimestampMsg / 1000);
+    localtime_r(&received, &stTime_receive);
+
+    //// callsign
+    if (mm->callsign_valid)
+        p += sprintf(p, "\"Tail\":\"%s\",", mm->callsign);
+    else
+        p += sprintf(p, "\"Tail\":null,");
+
+    //// altitude & gnss
+    bool alt_is_geom = false;
+    if (Modes.use_gnss && mm->altitude_geom_valid) {
+        p += sprintf(p, "\"Alt\":%d,",mm->altitude_geom);
+        alt_is_geom = true;
+    } else if (mm->altitude_baro_valid)
+        p += sprintf(p, "\"Alt\":%d,",mm->altitude_baro);
+    else
+        p += sprintf(p, "\"Alt\":null,");
+
+    // altitude source
+    if (alt_is_geom)
+        p += sprintf(p, "\"AltIsGNSS\":true,");
+    else
+        p += sprintf(p, "\"AltIsGNSS\":false,");
+
+    // GNSS alt. delta From baro alt.
+    if (trackDataValid(&a->geom_delta_valid))
+        p += sprintf(p, "\"GnssDiffFromBaroAlt\":%d,",a->geom_delta);
+    else
+        p += sprintf(p, "\"GnssDiffFromBaroAlt\":null,");
+    ////
+
+    //// ground speed and track
+    if (mm->gs_valid)
+        p += sprintf(p, "\"Speed_valid\":true,\"Speed\":%.0f,", mm->gs.selected);
+    else
+        p += sprintf(p, "\"Speed_valid\":false,\"Speed\":null,");
+
+    //// ground heading
+    if (mm->heading_valid && mm->heading_type == HEADING_GROUND_TRACK)
+        p += sprintf(p, "\"Track\":%.0f,", mm->heading);
+    else
+        p += sprintf(p, "\"Track\":null,");
+
+    //// position
+    if (mm->cpr_decoded)
+        p += sprintf(p, "\"Lat\":%.6f,\"Lng\":%.6f,\"Position_valid\":true,",
+                    mm->decoded_lat, mm->decoded_lon);
+    else
+        p += sprintf(p, "\"Lat\":null,\"Lng\":null,\"Position_valid\":false,");
+
+    //// vrate
+    if (Modes.use_gnss) {
+        if (mm->geom_rate_valid)
+            p += sprintf(p, "\"Vvel\":%d,", mm->geom_rate);
+        else if (mm->baro_rate_valid)
+            p += sprintf(p, "\"Vvel\":%d,", mm->baro_rate);
+        else
+            p += sprintf(p,  "\"Vvel\":null,");
+    } else {
+        if (mm->baro_rate_valid)
+            p += sprintf(p, "\"Vvel\":%d,", mm->baro_rate);
+        else if (mm->geom_rate_valid)
+            p += sprintf(p, "\"Vvel\":%d,", mm->geom_rate);
+        else
+            p += sprintf(p,  "\"Vvel\":null,");
+    }
```

Comment:
I assume that stratux wants either one behaviour or the other here (use_gnss comes from a command line option) so you can probably remove one of these branches